src/config.py

```
@@ -118,24 +120,24 @@ def load_config(self) -> None:
             exit(1)
 
         # 2. Load User Overrides
         if os.path.exists(USER_CONFIG_FILE):
             try:
                 with open(USER_CONFIG_FILE, encoding="utf-8") as f:
                     user_overrides = yaml.safe_load(f) or {}
-                self.deep_merge(raw_config, user_overrides)
+                self.deep_merge(raw_config, user_overrides, replace_keys={"models"})
             except Exception as e:
-                logging.error(f"Error loading config.yaml: {e}")
+                logging.exception(f"Error loading config.yaml: {e}")
```

Ruff (0.14.10)

`exception()` call on root logger

(LOG015)